app/models/hostgroup.rb

```diff
@@ -212,6 +213,12 @@ def update_ancestry_puppetclasses
     unscoped_find(ancestry.to_s.split('/').last.to_i).update_puppetclasses_total_hosts if ancestry.present?
   end
 
+  def children_hosts_count
+    count = hosts_count
+    return count unless children.any?
+    count += children.map(&:children_hosts_count).reduce(:+)
```

Comment:
`reduce` can accept a starting value that is added to, in this case count. It is also safe to run `map` & `reduce` on `children` even if there are none since `children` returns an empty array in that case.